cli/cppcheckexecutor.cpp

```diff
@@ -512,6 +512,11 @@ void StdLogger::writeCheckersReport()
         if (fout.is_open())
             fout << checkersReport.getReport(mCriticalErrors);
     }
+
+    if (mSettings.xml && mSettings.xml_version == 3) {
+        reportErr("    </errors>\n");
```

Comment:
this looks shaky.. the `</errors>` is written if version is 3 but not if it's version 2.

I rather write it here always so we know later that it's written.

Reply:
yes but it may not be written if we don't have checkers active so

Reply:
imho the ErrorLogger::getXmlHeader always writes the `<errors>` so we can always write `</errors>` here.

    if (mSettings.xml) {
        reportErr("    </errors>\n");
        if (mSettings.xml_version == 3)
            reportErr(checkersReport.getXmlReport(mCriticalErrors));
    }